platform/darwin/src/MGLShapeSource.mm

```diff
@@ -56,6 +56,16 @@ - (instancetype)initWithIdentifier:(NSString *)identifier shape:(nullable MGLSha
     return self;
 }
 
+- (instancetype)initWithIdentifier:(NSString *)identifier features:(NS_ARRAY_OF(MGLFeature) *)features options:(nullable NS_DICTIONARY_OF(MGLShapeSourceOption, id) *)options {
+    MGLShapeCollectionFeature *shapeCollectionFeature = [MGLShapeCollectionFeature shapeCollectionWithShapes:features];
```

Comment:
Should we check the type of each object in `features` and log a one time warning if a non-feature makes it through?